### PR TITLE
[Testing] Fix races in BlockState

### PR DESCRIPTION
**Pull request by SaveTheRbtz**
Ref: #3034

### CODE REVIEW

**Review by zhangchiqing — APPROVED** [review]

File: integration/tests/lib/block_state.go [hunk 1]
```diff
@@ -136,7 +136,9 @@ func (bs *BlockState) processAncestors(t *testing.T, b *messages.BlockProposal,
 // latest finalized block is 10, this will wait until any block height higher than 10 is finalized, and
 // returns that block.
 func (bs *BlockState) WaitForHighestFinalizedProgress(t *testing.T) *messages.BlockProposal {
```
Comment (pattyshack) [inline integration/tests/lib/block_state.go:138]: This function seems like a flaky test generator.  In particular, currentFinalized could be have been updated even before WaitForHighestFinalizedProgress got executed.  We should pass in the currentFinalized explicitly.

File: integration/tests/lib/block_state.go [hunk 2]
```diff
@@ -136,7 +136,9 @@ func (bs *BlockState) processAncestors(t *testing.T, b *messages.BlockProposal,
...
+	bs.RLock()
```
Comment (pattyshack) [inline integration/tests/lib/block_state.go:139]: per usual, use defer unlock pattern plz (refactor into a help method if necessary)

File: integration/tests/lib/block_state.go (hunk 1 above)
Comment (SaveTheRbtz) [reply integration/tests/lib/block_state.go:138]: Yeah, it seemed quite bad for me at first glance usages are quite legit: it is used as a healthcheck for the network and bootstraps the `currentFinalized` block that is used thereafter in tests.

Comment (SaveTheRbtz) [reply integration/tests/lib/block_state.go]: Renamed to `WaitForInitialization` to highlight its intended usage.

**Review by pattyshack — APPROVED** [review]

**Comment by SaveTheRbtz** [comment]
bors merge

**Pull request merged**